lib/Doctrine/ORM/Query/QueryException.php

```diff
@@ -116,6 +94,24 @@ public static function unknownParameter($key)
     {
         return new self('Invalid parameter: token ' . $key . ' is not defined in the query.');
     }
+    
+    /**
+     * @param array $expected
+     * @param array $received
+     *
+     * @return QueryException
+     */
+    public static function missingParameter($expected, $received)
+    {
+        foreach (array_keys($expected) as $token) {
+            if (!in_array($token, $received)) {
+                return new self('Missing parameter: no parameter set for token ' . $token . '.');
+            }
+        }
+        
+        return new self('Too few parameters: the query defines ' . count($expected)
+                . ' parameters but you only bound ' . count($received));
```

Comment:
Would it be feasible to make an exception that uses a `json_serialize` of expected vs passed parameters (plus the diff)?

Reply:
To clarify, from a dev perspective, if I forget 2 params and do a "edit and refresh until it works", this kind of exception on a single parameter means multiple refreshes on multiple mistakes

Reply:
@Ocramius can you elaborate on what you mean by `json_serialize`?

Reply:
I'm confused, do you mean `json_encode` or are you really referring to [JsonSerializable](http://php.net/manual/en/jsonserializable.jsonserialize.php)?

Reply:
Yeah, sorry, `json_encode()` - PHP's awesome naming.